pyvo/dal/tests/test_datalink.py

```diff
@@ -144,21 +144,14 @@ def test_datalink():
     results = vo.spectrumsearch(
         'http://example.com/ssa_datalink', (30, 30))
 
-    datalinks = next(results.iter_datalinks())
+    for preserve_order in [False, True]:
+        datalinks = next(results.iter_datalinks(preserve_order=preserve_order))
```

Comment:
Given this is a test for identical functionality, I guess I would exhaust the iterator, perhaps with something like set(itertools.chain(...)) (probably converting the links to tuples of (access_url, semantics,  description) or so).